### DISCUSSION TITLE
When running HammerDB-3.3 TPCH for SQL Server 2019 ,it occurs error

### DISCUSSION THREAD

**Discussion by Wendy798** [TPC-Council/HammerDB#87] [Q&A] [answered]
Following the Hammerdb  guide，I know Hammerdb does‘t  support SQL server 2019 now（https://www.hammerdb.com/docs/ch01s02.html）.For a simple evaluation, I try it for SQL Server 2019 , but it always occurs  error，details as  follow：
1.Test environment ：OS redhat 7.6， DB： SQL Server 2019 ，Hammerdb TPCH Scale=1000
2. Exception：
1)  the amount of  data is 2200GB ，which is far beyond the 1000GB
2)  see the hammerdb.log ，no warning，no error and it prints completed 
3）login  database，run "select count(*) from lineitem", it  shows “Arithmetic overflow error converting expression to data type int.”    And the table “orders ” only has one column "O_ORDERDATE" but the row count is correct as 1500000000.
4）I try Hammerdb TPCH Scale=10 for SQL Server 2019，it can load and run correctly as expectation. But only  Scale=1000,  it occurs errors after commpleting schema build

**Comment by sm-shaw** [comment, marked as answer]
Although SQL Server 2019 is not listed as tested there is no known reason why it should not work correctly.  Yes you would expect the dataset to be larger than 1000GB for SF1000 as you have to allow for indexes etc - in a quick test with SF1 the datafile is 2.25GB.
The error “Arithmetic overflow error converting expression to data type int.” is an error in the query rather than the database - in this case you use the int count rather than count_big that returns bigint that you should be using. (This is used in the queries eg select o_orderpriority, count_big(*) as order_count  ...)
For the error  “orders ” only has one column "O_ORDERDATE" but the row count is correct - this is inconsistent with the build having completed correctly and the row count correct - the build would error if there are columns missing and would be unable to create the foreign key on O_ORDERKEY. Try connecting with SSMS to verify whether columns really are missing.

**Comment by sm-shaw** [comment]
No further update, closing as error in querying the dataset build by HammerDB rather than the build itself.
